src/main/java/org/folio/rest/impl/GetOrdersByIdHelper.java

```diff
@@ -0,0 +1,250 @@
+          });
+      }).exceptionally(t -> {
+        logger.error("Failed to build composite purchase order", t.getCause());
+        future.completeExceptionally(t.getCause());
+        return null;
+      });
+
+    return future;
+  }
+
+  private CompletableFuture<JsonObject> getPurchaseOrder(String id, String lang) {
+    CompletableFuture<JsonObject> future = new VertxCompletableFuture<>(ctx);
+
+    try {
+      httpClient.request(HttpMethod.GET, String.format("/purchase_order/%s?lang=%s", id, lang), okapiHeaders)
+        .thenApply(HelperUtils::verifyAndExtractBody)
+        .thenAccept(future::complete)
+        .exceptionally(t -> {
+          logger.error("Exception calling GET /purchase_order/" + id, t);
+          future.completeExceptionally(t);
+          return null;
+        });
+    } catch (Exception e) {
+      logger.error("Exception calling GET /purchase_order/" + id, e);
+      future.completeExceptionally(e);
+    }
+
+    return future;
+  }
+
+  private CompletableFuture<List<PoLine>> getPoLines(String id, String lang) {
+    CompletableFuture<List<PoLine>> future = new VertxCompletableFuture<>(ctx);
+
+    try {
+      httpClient.request(HttpMethod.GET,
+          String.format("/po_line?limit=999&query=purchase_order_id==%s&lang=%s", id, lang), okapiHeaders)
+        .thenApply(HelperUtils::verifyAndExtractBody)
+        .thenAccept(body -> {
+          List<PoLine> lines = new ArrayList<>();
+          List<CompletableFuture<Void>> futures = new ArrayList<>();
+
+          for (int i = 0; i < body.getJsonArray("po_lines").size(); i++) {
+            JsonObject line = body.getJsonArray("po_lines").getJsonObject(i);
+            futures.add(resolvePoLine(line)
+              .thenAccept(lines::add));
+          }
+
+          VertxCompletableFuture.allOf(ctx, futures.toArray(new CompletableFuture[futures.size()]))
+            .thenAccept(v -> future.complete(lines))
+            .exceptionally(t -> {
+              future.completeExceptionally(t.getCause());
+              return null;
+            });
+        })
+        .exceptionally(t -> {
+          logger.error("Exception gathering po_line data:", t);
+          throw new CompletionException(t);
+        });
+    } catch (Exception e) {
+      logger.error("Exception calling GET /po_line/" + id, e);
+      future.completeExceptionally(e);
+    }
+
+    return future;
+  }
+
+  private CompletableFuture<PoLine> resolvePoLine(JsonObject line) {
+    CompletableFuture<PoLine> future = new VertxCompletableFuture<>(ctx);
+
+    List<CompletableFuture<Void>> futures = new ArrayList<>();
+    futures.add(resolveSubObjIfPresent(line, "adjustment", "/adjustment/"));
+    futures.add(resolveSubObjIfPresent(line, "cost", "/cost/"));
+    futures.add(resolveSubObjIfPresent(line, "details", "/details/"));
+    futures.add(resolveSubObjIfPresent(line, "eresource", "/eresource/"));
+    futures.add(resolveSubObjIfPresent(line, "location", "/location/"));
```

Comment:
Isn't this URL /locations? https://github.com/folio-org/mod-inventory-storage/blob/master/ramls/location.raml , or Am I looking the wrong API?

Reply:
nope I think you're correct.  I had a feeling I was going to get burned by the competing plural/singular api conventions eventually